### PR TITLE
Add .insn directive

### PR DESCRIPTION
**Pull request by charlie-rivos**
Add .insn directive with brief description and link to full documentation.

Co-developed-by: Tim Hutt <timothy.hutt@codasip.com>
Co-developed-by: Christoph Müllner <christoph.muellner@vrull.eu>

### CODE REVIEW

**Comment by charlie-rivos** [comment]
@cmuellner if you think it is valuable I can add all of the instruction formats, or we can leave it as-is and link to the Binutils docs

File: src/asm-manual.adoc [hunk 1]
```diff
@@ -176,6 +176,7 @@ The following table lists assembler directives:
 |.zero        | integer                        | zero bytes
 |.variant_cc  | symbol_name                    | annotate the symbol with variant calling convention
 |.attribute   | name, value                    | RISC-V object attributes, more detailed description see <<.attribute, .attribute>>.
+| insn        | see description                | emit a custom instruction encoding, see <<.insn, .insn>>
```
Comment (cmuellner) [inline src/asm-manual.adoc:179]: Leading `.` (dot) is missing

**Review by cmuellner — COMMENTED** [review]
I think that's a good start.
If we decide later to standardize more here, then we can easily extend.
But for now the scope looks reasonable to me.

**Comment by cmuellner** [comment]
CC: @asb @kito-cheng @Timmmm

File: src/asm-manual.adoc (hunk 1 above)
Comment (charlie-rivos) [reply src/asm-manual.adoc]: Fixed!

**Comment by Timmmm** [comment]
Did you mean to leave a .patch file in there?

**Comment by charlie-rivos** [comment]
> Did you mean to leave a .patch file in there?

Whoops, thank you for catching that, I removed it

**Review by Timmmm — APPROVED** [review]

**Review by cmuellner — APPROVED** [review]
LGTM

**Review by kito-cheng — APPROVED** [review]
LGTM as well:)

**Comment by cmuellner** [comment]
Closing this as #109 was merged.

**Pull request merged**